langchain/src/smith/runner_utils.ts

```diff
@@ -49,9 +49,13 @@ class RunIdExtractor {
   handleChainStart = (
     _chain: Serialized,
     _inputs: ChainValues,
-    runId: string
+    runId: string,
+    parentRunId?: string
   ) => {
-    this.runIdPromiseResolver(runId);
+    if (!parentRunId) {
+      // Only resolve root runs
+      this.runIdPromiseResolver(runId);
```

Comment:
Also added this. Not strictly required, but I think it makes sense to only match on the root chain runs